### PR TITLE
Update how time and date is handled.

### PR DESCRIPTION
**Pull request by vpetrovykh**
`std::time` is now `std::naivetime` and is no longer time zone aware.

The function `current_time` is now `local_time`.

### CODE REVIEW

File: edb/lib/std/30-datetimefuncs.eql [hunk 1]
```diff
@@ -28,9 +28,9 @@ std::current_date() -> std::date
 
 
 CREATE FUNCTION
-std::current_time() -> std::time
+std::local_time() -> std::naivetime
```
Comment (1st1) [inline edb/lib/std/30-datetimefuncs.eql]: I propose to change the order to `time_local()`; I'd like all time-related functions to be grouped together.

Comment (vpetrovykh) [reply edb/lib/std/30-datetimefuncs.eql]: We have:
- `current_date`
- `current_datetime`
- `local_time`

Flipping the order seems a bit counter-intuitive.

Comment (elprans) [reply edb/lib/std/30-datetimefuncs.eql]: I vote for `current_localtime()`

**Comment by vpetrovykh** [comment]
I've pushed a revised version of the update with new time and date functions.

File: edb/lang/schema/scalars.py [hunk 2]
```diff
@@ -232,7 +232,9 @@ def _is_assignment_castable_impl(source: str, target: str) -> bool:
         ('std::str', 'std::str', 'std::str'),
         ('std::bytes', 'std::bytes', 'std::bytes'),
         ('std::datetime', 'std::timedelta', 'std::datetime'),
-        ('std::time', 'std::timedelta', 'std::time'),
+        ('std::naivedatetime', 'std::timedelta', 'std::naivedatetime'),
```
Comment (1st1) [inline edb/lang/schema/scalars.py]: `std::naivedatetime` -> `std::naive_datetime`? @elprans

File: edb/lib/std/30-datetimefuncs.eql [hunk 3]
```diff
@@ -21,21 +21,62 @@
...
+std::datetime_now(mode: std::str='') -> std::datetime
 {
-    FROM SQL 'SELECT current_date';
+    FROM SQL $$
+    SELECT (
+        CASE WHEN "mode" = 'transaction' THEN
+            transaction_timestamp()
+        WHEN "mode" = 'statement' THEN
+            statement_timestamp()
+        ELSE
```
Comment (1st1) [inline edb/lib/std/30-datetimefuncs.eql]: I don't like that this function can accept any garbage.  I also think that having more than 2 different modes of operation is way too much for a single function. Let's discuss splitting this into three functions.

Comment (elprans) [reply edb/lib/std/30-datetimefuncs.eql]: `std::datetime_of_transaction`, `std::datetime_of_statement`, `std::datetime_now` seem like reasonable names to me.

Comment (vpetrovykh) [reply edb/lib/std/30-datetimefuncs.eql]: I've updated the PR.

File: edb/lib/std/30-datetimefuncs.eql [hunk 4]
```diff
@@ -21,21 +21,66 @@
 
 
 CREATE FUNCTION
-std::current_date() -> std::date
+std::datetime_now() -> std::datetime
```
Comment (elprans) [inline edb/lib/std/30-datetimefuncs.eql]: Perhaps, `datetime_current` would be a better name, making some distance from SQL `now()` which is a transaction timestamp.

Comment (vpetrovykh) [reply edb/lib/std/30-datetimefuncs.eql]: Sure, let's call it `datetime_current`. Avoiding confusion with SQL `now()` is important here.

**Pull request merged**